licensecheck/get_deps.py

```diff
@@ -138,27 +141,30 @@ def resolveExtraReq(extraReq: str) -> ucstr | None:
 
 	# Get Dependencies (1 deep)
 	requirementsWithDeps = reqs.copy()
+
+	def update_dependencies(dependency: str) -> None:
+		dep = resolveReq(dependency, False)
+		req = resolveReq(requirement, False)
+		extra = resolveExtraReq(dependency)
+		if extra is not None:
+			if req in extrasReqs and extra in extrasReqs.get(req, []):
+				requirementsWithDeps.add(dep)
+		else:
+			requirementsWithDeps.add(dep)
+
 	for requirement in reqs:
 		try:
 			pkgMetadata = metadata.metadata(requirement)
-			for req in [resolveReq(req) for req in pkgMetadata.get_all("Requires-Dist") or []]:
-				requirementsWithDeps.add(req)
+			for dependency in pkgMetadata.get_all("Require-Dist") or []:
+				update_dependencies(dependency)
```

Comment:
`info` Believe the extras filtering wasn't previously happening on this code path.